examples_utils/benchmarks/environment_utils.py

```diff
@@ -145,6 +147,11 @@ def infer_paths(args: ArgumentParser, benchmark_dict: dict) -> ArgumentParser:
 
     return args
 
+def get_git_commit_hash() -> str:
+    # assumed we're in the top level directory of the git repo
+    process = subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode(sys.stdout.encoding).strip()
```

Comment:
I think this needs to be expanded a bit to cater for:

- Making sure this runs in the correct folder
- Handle errors? If you are not running from inside a repo it needs to return something sensible

Reply:
Can you add a test to make sure? I have the suspicion that this will raise a `CalledProcessError`. I think you might need to use subprocess.run to get what you want. The test should be reasonably straightforward:

```
def test_git_commit_hash_in_repo():
    os.chdir(pathlib.Path(__file__).parent)
    hash = get_git_commit_has()
    # assert some condition


def test_git_commit_hash_out_of_repo(tmp_path):
    os.chdir(tmp_path)
    not_a_hash = get_git_commit_has()
    assert "Not a git repo" in not_a_hash

```